lightning/src/sync/debug_sync.rs

```diff
@@ -248,6 +262,13 @@ impl<T> Mutex<T> {
 	}
 }
 
+impl <T> LockTestExt for Mutex<T> {
```

Comment:
More of a general question, but how come rust-lightning has its own synchronization primitives like `Mutex` and `RwLock`?

Reply:
Just took a better look at the locks and that makes more sense now 👍